sw/device/silicon_creator/rom/doc/rom_overview_specification.md

```diff
@@ -0,0 +1,272 @@
+
+See the [[startup]](#startup) specification for details.
+
+
+### ROM.ASM-SRAM-INIT
+
+
+
+*   Turn on the minimum level of entropy required to initialize memory scrambling.
+*   Enable SRAM scrambling and initialize SRAM.
+
+See the [[startup]](#startup) specification for details.
+
+
+### ROM.ASM-PMP-INIT
+
+
+
+*   Initialize the PMP.  Basic configuration:
+    *   R+X for ROM regions.
+    *   R for FLASH
+    *   RW for RAM
+
+See the [[memory protection]](memory_protection.md) specification for details.
+
+
+### ROM.ASM-CRT-INIT
+
+
+
+*   Zero .bss and copy .data to RAM.
+*   Initialize stack pointer and global pointer.
+*   Load the interrupt vector into mtvec.
+*   Jump to `mask\_rom\_main`.
+
+See the [[startup]](#startup) specification for details.
+
+
+### ROM.WATCHDOG-INIT
+
+
+
+*   Initialize the watchdog timer to instigate a reset after $timeout.  (should we read this value from OTP?)
+
+
+### ROM.RST-REASON-INIT
+
+
+
+*   Read the reset reason.  If power-on reset, init the retention RAM.
+
+
+### ROM.SEC-MMIO-INIT
+
+
+
+*   Initialize the sec\_mmio library.  Rescan any OTPs read prior to this point to load the expectations into sec\_mmio’s register tracking.
+
+
+### ROM.PMP-INIT
+
+
+
+*   Initialize the PMP module with values from ROM.ASM-PMP-INIT.
+
+See the [[memory protection]](memory_protection.md) specification for details.
+
+
+### ROM.SHUTDOWN-INIT
+
+
+
+*   Initialize the shutdown module with values from OTP.
+*   Loads alert configurations and alert class configurations from OTP into the alert handler block.
+
+See the [[shutdown]](#shutdown) specification for details.
+
+
+### ROM.PINMUX-INIT
+
+
+
+*   Initialize pinmux to allow uart output and reading of the strap pins.
+
+See the [[pin configuration]](#pinconfig) specification for details.
+
+
+### ROM.UART-INIT
+
+
+
+*   Initialize the uart to 115200 bps.
+*   Print the OpenTitan version identifier (e.g. git version of the mask rom code).
+*   Print the lifecycle state.
+
+
+### ROM.FLASH-INIT
+
+
+
+*   Flash initializes its own scrambling parameters in hardware.  Make sure the init has finished by waiting for FLASH\_CTRL.STATUS.INIT\_WIP to be zero.
+
+
+### ROM.BOOTSTRAP-MODE
+
+
+
+*   Read the value of the strapping pins.
+    *   If they indicate bootstrap mode and if bootstrap is permitted (via OTP ROM\_BOOTSTRAP\_EN), perform the bootstrap protocol as specified in the [Bootstrap Spec].
+    *   For all other values, continue booting without bootstrap mode.
+
+See the [[bootstrap]](#bootstrap) specification for details.
+
+
+### ROM.KEYMGR-INIT
+
+
+
+*   Initialize the key manager.
+
+
+### ROM.BOOT-POLICY-READ
+
+
+
+*   Read the boot policy from the flash INFO page and retention RAM and determine the parameters which will control booting the ROM\_EXT (which side to boot, which side is fallback, temporary side override and any anti-rollback parameters.
+
+See the [[bootpolicy]](#bootpolicy) specification for details.
```

Comment:
We will probably use this logic in ROM_EXT. ROM prioritizes the slot with the newest (in terms of the security version) ROM_EXT.

Reply:
What is the current behavior here?  Read the boot policy and make sure the selected ROM_EXT meets the minimum version requirement from the policy?